src/core/reference/include/openvino/reference/topk.hpp

```diff
@@ -103,5 +105,39 @@ void topk(const T* arg,
         }
     }
 }
+
+/**
+ * @brief Reference implementation for TopK operator
+ *
+ * @param arg         Pointer to input data.
+ * @param out_indices Pointer to output indicies.
+ * @param out_values  Pointer to output values.
+ * @param in_shape    Input data shape.
+ * @param out_shape   Output data (values, indicies) shape.
+ * @param axis        Axis for search of top K elements.
+ * @param k           Number to find of top elements.
+ * @param compute_max Select mode of find max or min.
+ * @param sort        Sorting type.
+ */
+template <typename T,
+          typename U,
+          typename std::enable_if<!std::is_same<typename std::decay<U>::type, int64_t>::value>::type* = nullptr>
+void topk(const T* arg,
+          U* out_indices,
+          T* out_values,
+          const Shape& in_shape,
+          const Shape& out_shape,
+          const size_t axis,
+          const size_t k,
+          const bool compute_max,
+          const op::TopKSortType sort = op::TopKSortType::NONE) {
+    const auto out_count = shape_size(out_shape);
+    auto temp_out_indices = std::vector<int64_t>(out_count);
+
+    topk(arg, temp_out_indices.data(), out_values, in_shape, out_shape, axis, k, compute_max, sort);
+
+    for (auto i = out_count; i-- > 0;)
```

Comment:
```suggestion
    for (auto i = out_count; i > 0; --i)
```
?

Reply:
It won't work as expected - decrement is needed before loop body execution.